packages/product/src/services/product-module-service.ts

```diff
@@ -450,8 +456,28 @@ export default class ProductModuleService<
     data: ProductTypes.UpdateProductOptionDTO[],
     @MedusaContext() sharedContext: Context = {}
   ) {
-    const productOptions = await this.productOptionService_.update(
-      data,
+    const normalizedInput = data.map((opt) => {
+      return {
+        ...opt,
+        ...(opt.values
+          ? {
+              values: opt.values.map((v) => {
+                return typeof v === "string" ? { value: v } : v
+              }),
+            }
+          : {}),
+      } as ProductTypes.UpdateProductOptionDTO
+    })
+
+    if (normalizedInput.some((option) => !option.id)) {
+      throw new MedusaError(
+        MedusaError.Types.INVALID_DATA,
+        "Tried to update options without specifying an ID"
+      )
+    }
```

Comment:
question: Could this be moved to before the normalization?

Reply:
Yeah we probably can, I'll do more cleanup when we have more focus on the quality of the modules